src/contextify.cc

```diff
@@ -126,27 +128,28 @@ class ContextifyContext : public ObjectWrap {
 class ContextWrap : public ObjectWrap {
 public:
     static void Init(void) {
-        NanScope();
-        Local<FunctionTemplate> tmpl = NanNew<FunctionTemplate>();
+        Nan::HandleScope scope;
+
+        Local<FunctionTemplate> tmpl = Nan::New<FunctionTemplate>();
         tmpl->InstanceTemplate()->SetInternalFieldCount(1);
-        NanAssignPersistent(functionTemplate, tmpl);
-        NanAssignPersistent(constructor, tmpl->GetFunction());
+        functionTemplate.Reset(tmpl);
+        constructor.Reset(tmpl->GetFunction());
     }
 
     static Local<Context> createV8Context(Handle<Object> jsContextify) {
-        NanEscapableScope();
-        Local<Object> wrapper = NanNew(constructor)->NewInstance();
+        Nan::EscapableHandleScope scope;
+        Local<Object> wrapper = Nan::New(constructor)->NewInstance();
 
         ContextWrap *contextWrapper = new ContextWrap();
         contextWrapper->Wrap(wrapper);
 
-        Local<Object> obj = NanNew(jsContextify);
-        NanMakeWeakPersistent(obj, contextWrapper, &weakCallback);
+        Local<Object> obj = Local<Object>::New(Isolate::GetCurrent(), jsContextify);
```

Comment:
Hmm.. I should check if this Object constructor with an argument is something new or if I've simply missed it.

On November 3, 2015 10:30:11 AM GMT+02:00, Ido Ben-Yair <notifications@github.com> wrote:
>>  
>>          ContextWrap *contextWrapper = new ContextWrap();
>>          contextWrapper->Wrap(wrapper);
>>  
>> -        Local<Object> obj = NanNew(jsContextify);
>> -        NanMakeWeakPersistent(obj, contextWrapper, &weakCallback);
>> +        Local<Object> obj =
>Local<Object>::New(Isolate::GetCurrent(), jsContextify);
>
>clang says:
>
>```
>In file included from ../src/contextify.cc:3:
>In file included from ../node_modules/nan/nan.h:200:
>../node_modules/nan/nan_new.h:208:31: error: too many arguments to
>function call, expected 0, have 1
>  return imp::Factory<T>::New(arg0);
>         ~~~~~~~~~~~~~~~~~~~~ ^~~~
>../src/contextify.cc:146:34: note: in instantiation of function
>template specialization 'Nan::New<v8::Object, v8::Local<v8::Object> >'
>requested here
>        Local<Object> obj = Nan::New<Object>(jsContextify);
>                                 ^
>../node_modules/nan/nan_implementation_12_inl.h:181:1: note: 'New'
>declared here
>Factory<v8::Object>::return_t
>^
>1 error generated.
>```
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/brianmcd/contextify/pull/190/files#r43724336